service/raft/group0_state_machine.cc

```diff
@@ -189,6 +189,7 @@ future<> group0_state_machine::merge_and_apply(group0_state_machine_merger& merg
     },
     [&] (topology_change& chng) -> future<> {
         auto tablet_keys = replica::get_tablet_metadata_change_hint(chng.mutations);
+        co_await _ss.validate_tablets_transition(tablet_keys);
```

Comment:
This is too late, if validation fails here, group0 will be killed. We should validate before committing to group0.

Reply:
OK, moved the check to group0_client::prepare_command()